### DISCUSSION TITLE
How do I ensure quarto places output file in a specific directory?

### DISCUSSION THREAD

**Discussion by rkaramc** [quarto-dev/quarto-cli#11131] [Q&A]
### Description

Trying to generate a PDF document for download through Shiny for Python.

Sequence is:
1. Generate data in Shiny for Python
2. Generate `tmpfile.qmd` using Jinja in `/tmp/tmpfolder`
3. Generate `tmpfile.pdf` using Quarto
   * Call `render()` in `quarto-dev/quarto-python` with these args:
     - `input=/tmp/tmpfolder/tmpfile.qmd` (input)
     - `output_format=pdf`
     - `output_file=os.path.basename(tmpfile.pdf)`    -- quarto complains if name contains path
     - `output_dir=os.path.dirname(tmpfile.pdf)`        -- expecting file to be placed in this directory; see [PR](https://github.com/quarto-dev/quarto-python/pull/7)
     - `execute=false`                                                  -- didn't want to re-create the shiny app in qmd, data is already in qmd
5. Download `/tmp/tmpfolder/tmpfile.pdf` to client browser using Shiny for Python

Quarto (or pandoc?) places the output file in the current working directory. This happens to be the folder where I am running the shiny app (`shiny run app:app`).

How do I get quarto (or pandoc?) to use `/tmp/tmpfolder`? Or should I just move the file over using Python?

Thanks

**Comment by mcanouil** [comment]
Did you close (as duplicate) on purpose?
